providers/implementations/kdfs/pbkdf2.c

```diff
@@ -317,22 +310,14 @@ static int kdf_pbkdf2_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     }
 
     if (p.pw != NULL && !pbkdf2_set_membuf(&ctx->pass, &ctx->pass_len, p.pw))
-            return 0;
+        return 0;
 
-    if (p.salt != NULL) {
-        if (!lower_bound_check_passed(ctx, (int)p.salt->data_size, UINT64_MAX, SIZE_MAX,
```

Comment:
Producing the error when trying to set something too short seems reasonable to me.  Likewise, insisting that the bypass is set before setting the salt also seems reasonable.

Reply:
So would you prefer this just to be a documentation change? To me this code is a horrible mess.

Reply:
> So would you prefer this just to be a documentation change? To me this code is a horrible mess.

I would like this question answered before I do anything with this PR.

Reply:
More than a documentation change.  I'd like to see the bounds check and error return in the set params still.